### PR TITLE
Go modules

### PR DESCRIPTION
**Pull request by livingsilver94**
- Removed git submodules in favor of Go modules
- Moved src/ content to the root, as it is idiomatic in Go projects
- Swapped some libraries forked by Josh with their upstream versions
- Updated `github.com/cheggaaa/pb` to v3. Some API changed, so I updated the code. No aesthetic edits were deliberately made. The library changed how it renders the bar though.
---
- Simplified the Makefile
---
- Updated the copyright strings

### CODE REVIEW

**Comment by livingsilver94** [comment]
My PR leaves the README in a wrong state. Should I update it too?

**Review by DataDrake — CHANGES_REQUESTED** [review]
As far as I can see, this is OK. But the official copyright should be `Solus Project <copyright@getsol.us>` since we setup an email for copyright issues.

@JoshStrobl will need to take a closer look at the deps since he probably forked them to fix an upstream issue that was breaking our workflow.

I have mixed feelings about the import ordering since we usually rely on gofmt, not goimports. Personally, I sort them alphabetically unless I alias a package, in which case I generally follow the alias instead of the package it refers to because it's easier to keep sorted. The goimports style feels like a hold-over from someone's C++ style guide where they wanted a clear separation between the standard library, project headers, and third-party headers. I don't think that's really necessary for us since our code is always hosted on a Git server, making our project-level imports distinctly belong to one of our namespace.

**Comment by livingsilver94** [comment]
Copyright strings fixed.

File: Makefile [hunk 1]
```diff
@@ -1,64 +1,55 @@
...
-	install -m 00644 man/*.5 $(DESTDIR)/usr/share/man/man5/.;
-
-
-ensure_modules:
-	@ ( \
-		git submodule init; \
-		git submodule update; \
-	);
+VERSION := 1.4.5.2
+EXENAME := solbuild
```
Comment (JoshStrobl) [inline Makefile]: Would prefer this be changed to BINARY or BINNAME than EXENAME

File: builder/build.go [hunk 2]
```diff
@@ -19,10 +19,11 @@ package builder
...
 	"errors"
 	"fmt"
-	log "github.com/Sirupsen/logrus"
-	"github.com/solus-project/libosdev/disk"
 	"os"
 	"path/filepath"
+
+	"github.com/getsolus/libosdev/disk"
+	log "github.com/sirupsen/logrus"
 )
```
Comment (JoshStrobl) [inline builder/build.go]: import order is a regression. Please use gofmt not goimports.

File: builder/chroot.go [hunk 3]
```diff
@@ -18,9 +18,10 @@ package builder
 
 import (
 	"fmt"
-	log "github.com/Sirupsen/logrus"
-	"github.com/solus-project/libosdev/commands"
 	"os"
+
+	"github.com/getsolus/libosdev/commands"
+	log "github.com/sirupsen/logrus"
 )
```
Comment (JoshStrobl) [inline builder/chroot.go]: import order is a regression. Please use gofmt not goimports.

File: builder/config.go [hunk 4]
```diff
@@ -18,18 +18,19 @@ package builder
 
 import (
 	"fmt"
-	"github.com/BurntSushi/toml"
 	"io/ioutil"
 	"os"
 	"path/filepath"
+
+	"github.com/BurntSushi/toml"
 )
```
Comment (JoshStrobl) [inline builder/config.go]: import order is a regression. Please use gofmt not goimports.

File: builder/copy.go [hunk 5]
```diff
@@ -17,11 +17,12 @@
...
 import (
-	log "github.com/Sirupsen/logrus"
-	"github.com/solus-project/libosdev/disk"
 	"io/ioutil"
 	"os"
 	"path/filepath"
+
+	"github.com/getsolus/libosdev/disk"
+	log "github.com/sirupsen/logrus"
 )
```
Comment (JoshStrobl) [inline builder/copy.go]: import order is a regression. Please use gofmt not goimports.

File: builder/eopkg.go [hunk 6]
```diff
@@ -18,13 +18,14 @@ package builder
...
-	"github.com/solus-project/libosdev/disk"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strings"
+
+	"github.com/getsolus/libosdev/commands"
+	"github.com/getsolus/libosdev/disk"
+	log "github.com/sirupsen/logrus"
 )
```
Comment (JoshStrobl) [inline builder/eopkg.go]: import order is a regression. Please use gofmt not goimports.

File: builder/history.go [hunk 7]
```diff
@@ -20,12 +20,13 @@ import (
 	"encoding/xml"
 	"errors"
 	"fmt"
-	"github.com/JoshStrobl/git2go"
 	"os"
 	"path/filepath"
 	"regexp"
 	"sort"
 	"time"
+
```
Comment (JoshStrobl) [inline builder/history.go]: import order is a regression. Please use gofmt not goimports.

File: builder/index.go [hunk 8]
```diff
@@ -19,10 +19,11 @@ package builder
 import (
 	"errors"
 	"fmt"
-	log "github.com/Sirupsen/logrus"
-	"github.com/solus-project/libosdev/disk"
 	"os"
 	"path/filepath"
+
+	"github.com/getsolus/libosdev/disk"
+	log "github.com/sirupsen/logrus"
```
Comment (JoshStrobl) [inline builder/index.go]: import order is a regression. Please use gofmt not goimports.

File: builder/manager.go [hunk 9]
```diff
@@ -18,15 +18,16 @@ package builder
...
 	"os"
 	"os/signal"
 	"path/filepath"
 	"strings"
 	"sync"
 	"syscall"
 	"time"
+
+	"github.com/getsolus/libosdev/disk"
+	log "github.com/sirupsen/logrus"
```
Comment (JoshStrobl) [inline builder/manager.go]: import order is a regression. Please use gofmt not goimports.

File: builder/namespaces.go [hunk 10]
```diff
@@ -17,8 +17,9 @@
 package builder
 
 import (
-	log "github.com/Sirupsen/logrus"
 	"syscall"
+
+	log "github.com/sirupsen/logrus"
 )
```
Comment (JoshStrobl) [inline builder/namespaces.go]: import order is a regression. Please use gofmt not goimports.

File: builder/overlay.go [hunk 11]
```diff
@@ -18,11 +18,12 @@ package builder
...
-	log "github.com/Sirupsen/logrus"
-	"github.com/solus-project/libosdev/commands"
-	"github.com/solus-project/libosdev/disk"
 	"os"
 	"path/filepath"
+
+	"github.com/getsolus/libosdev/commands"
+	"github.com/getsolus/libosdev/disk"
+	log "github.com/sirupsen/logrus"
 )
```
Comment (JoshStrobl) [inline builder/overlay.go]: import order is a regression. Please use gofmt not goimports.

File: builder/pkg.go [hunk 12]
```diff
@@ -17,14 +17,15 @@
...
 	"encoding/xml"
 	"errors"
 	"fmt"
-	"github.com/go-yaml/yaml"
 	"io/ioutil"
 	"os"
 	"strings"
+
+	"github.com/getsolus/solbuild/builder/source"
+	"gopkg.in/yaml.v2"
```
Comment (JoshStrobl) [inline builder/pkg.go]: import order is a regression. Please use gofmt not goimports.

File: builder/profile.go [hunk 13]
```diff
@@ -18,11 +18,12 @@ package builder
...
 import (
 	"fmt"
-	"github.com/BurntSushi/toml"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strings"
+
+	"github.com/BurntSushi/toml"
 )
```
Comment (JoshStrobl) [inline builder/profile.go]: import order is a regression. Please use gofmt not goimports.

File: builder/repos.go [hunk 14]
```diff
@@ -18,10 +18,11 @@ package builder
...
 import (
 	"fmt"
-	log "github.com/Sirupsen/logrus"
-	"github.com/solus-project/libosdev/disk"
 	"os"
 	"path/filepath"
+
+	"github.com/getsolus/libosdev/disk"
+	log "github.com/sirupsen/logrus"
 )
```
Comment (JoshStrobl) [inline builder/repos.go]: import order is a regression. Please use gofmt not goimports.

File: builder/source/git.go [hunk 15]
```diff
@@ -19,13 +19,14 @@ package source
...
-	"github.com/solus-project/libosdev/commands"
 	"net/url"
 	"os"
 	"path/filepath"
 	"strings"
+
+	"github.com/getsolus/libosdev/commands"
+	git "github.com/libgit2/git2go/v28"
+	log "github.com/sirupsen/logrus"
 )
```
Comment (JoshStrobl) [inline builder/source/git.go]: import order is a regression. Please use gofmt not goimports.

File: builder/source/simple.go [hunk 16]
```diff
@@ -21,13 +21,14 @@ import (
...
-	curl "github.com/JoshStrobl/go-curl"
-	"github.com/cheggaaa/pb"
 	"io/ioutil"
 	"net/url"
 	"os"
 	"path/filepath"
+
+	curl "github.com/andelf/go-curl"
+	"github.com/cheggaaa/pb/v3"
+	log "github.com/sirupsen/logrus"
```
Comment (JoshStrobl) [inline builder/source/simple.go]: import order is a regression. Please use gofmt not goimports.

File: builder/transit_manifest.go [hunk 17]
```diff
@@ -19,10 +19,11 @@ package builder
 import (
 	"bytes"
 	"errors"
-	"github.com/BurntSushi/toml"
 	"io/ioutil"
 	"path/filepath"
 	"strings"
+
+	"github.com/BurntSushi/toml"
```
Comment (JoshStrobl) [inline builder/transit_manifest.go]: import order is a regression. Please use gofmt not goimports.

File: builder/update.go [hunk 18]
```diff
@@ -17,10 +17,11 @@
 package builder
 
 import (
-	log "github.com/Sirupsen/logrus"
-	"github.com/solus-project/libosdev/disk"
 	"os"
 	"path/filepath"
+
+	"github.com/getsolus/libosdev/disk"
+	log "github.com/sirupsen/logrus"
```
Comment (JoshStrobl) [inline builder/update.go]: import order is a regression. Please use gofmt not goimports.

File: builder/userinfo.go [hunk 19]
```diff
@@ -18,12 +18,13 @@ package builder
...
 	"fmt"
-	log "github.com/Sirupsen/logrus"
-	"github.com/go-ini/ini"
 	"os"
 	"os/user"
 	"path/filepath"
 	"strconv"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/ini.v1"
```
Comment (JoshStrobl) [inline builder/userinfo.go]: import order is a regression. Please use gofmt not goimports.

File: builder/util.go [hunk 20]
```diff
@@ -31,6 +28,10 @@ import (
 	"strings"
 	"syscall"
 	"time"
+
+	"github.com/getsolus/libosdev/commands"
+	"github.com/getsolus/libosdev/disk"
+	log "github.com/sirupsen/logrus"
```
Comment (JoshStrobl) [inline builder/util.go]: import order is a regression. Please use gofmt not goimports.

File: cmd/build.go [hunk 21]
```diff
@@ -17,13 +17,14 @@
...
 	"fmt"
-	log "github.com/Sirupsen/logrus"
-	"github.com/spf13/cobra"
 	"os"
 	"strings"
+
+	"github.com/getsolus/solbuild/builder"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
 )
```
Comment (JoshStrobl) [inline cmd/build.go]: import order is a regression. Please use gofmt not goimports.

File: cmd/chroot.go [hunk 22]
```diff
@@ -17,13 +17,14 @@
...
 	"errors"
 	"fmt"
-	log "github.com/Sirupsen/logrus"
-	"github.com/spf13/cobra"
 	"os"
 	"strings"
+
+	"github.com/getsolus/solbuild/builder"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
```
Comment (JoshStrobl) [inline cmd/chroot.go]: import order is a regression. Please use gofmt not goimports.

File: cmd/delete_cache.go [hunk 23]
```diff
@@ -17,13 +17,14 @@
...
 	"fmt"
-	log "github.com/Sirupsen/logrus"
-	"github.com/spf13/cobra"
 	"os"
 	"strings"
+
+	"github.com/getsolus/solbuild/builder"
+	"github.com/getsolus/solbuild/builder/source"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
```
Comment (JoshStrobl) [inline cmd/delete_cache.go]: import order is a regression. Please use gofmt not goimports.

File: cmd/index.go [hunk 24]
```diff
@@ -17,12 +17,13 @@
...
-	"builder"
 	"fmt"
-	log "github.com/Sirupsen/logrus"
-	"github.com/spf13/cobra"
 	"os"
 	"strings"
+
+	"github.com/getsolus/solbuild/builder"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
```
Comment (JoshStrobl) [inline cmd/index.go]: import order is a regression. Please use gofmt not goimports.

File: cmd/init.go [hunk 25]
```diff
@@ -17,16 +17,16 @@
...
 	"os"
 	"strings"
 
-	log "github.com/Sirupsen/logrus"
-	"github.com/cheggaaa/pb"
-	"github.com/solus-project/libosdev/commands"
+	"github.com/cheggaaa/pb/v3"
+	"github.com/getsolus/libosdev/commands"
+	"github.com/getsolus/solbuild/builder"
+	log "github.com/sirupsen/logrus"
```
Comment (JoshStrobl) [inline cmd/init.go]: import order is a regression. Please use gofmt not goimports.

File: cmd/root.go [hunk 26]
```diff
@@ -17,9 +17,10 @@
 package cmd
 
 import (
-	"builder"
-	"github.com/spf13/cobra"
 	"os"
+
+	"github.com/getsolus/solbuild/builder"
+	"github.com/spf13/cobra"
```
Comment (JoshStrobl) [inline cmd/root.go]: import order is a regression. Please use gofmt not goimports.

File: cmd/update.go [hunk 27]
```diff
@@ -17,12 +17,13 @@
...
-	"builder"
 	"fmt"
-	log "github.com/Sirupsen/logrus"
-	"github.com/spf13/cobra"
 	"os"
 	"strings"
+
+	"github.com/getsolus/solbuild/builder"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
```
Comment (JoshStrobl) [inline cmd/update.go]: import order is a regression. Please use gofmt not goimports.

File: cmd/version.go [hunk 28]
```diff
@@ -18,6 +18,7 @@ package cmd
 
 import (
 	"fmt"
+
```
Comment (JoshStrobl) [inline cmd/version.go]: Ew no.

File: main.go [hunk 29]
```diff
@@ -17,10 +17,11 @@
...
 
 import (
-	_ "builder"
-	log "github.com/Sirupsen/logrus"
 	"os"
-	"solbuild/cmd"
+
+	_ "github.com/getsolus/solbuild/builder"
+	"github.com/getsolus/solbuild/cmd"
+	log "github.com/sirupsen/logrus"
```
Comment (JoshStrobl) [inline main.go]: import order is a regression. Please use gofmt not goimports.

**Review by JoshStrobl — CHANGES_REQUESTED** [review]

**Comment by JoshStrobl** [comment]
With regards to my fork, that won't be necessary thanks to:

1. Go vendors
2. The dedicated v28 branch for 0.28.x support.

My fork was performed to introduce fixes that eventually landed upstream, so I can drop that in favor of upstream and we can work on moving it away from 0.28 (solbuild is really only ever the blocker for libgit2 update).

I agree that the use of goimports instead of gofmt + our typical ordering is a regression and should be reverted.

**Comment by livingsilver94** [comment]
I sold my soul to the devil and used gofmt to sort imports.

**Review by JoshStrobl — APPROVED** [review]
LGTM to me now.

**Comment by DataDrake** [comment]
LGTM. Thanks!

**Pull request merged**